src/Command.hs

```diff
@@ -17,60 +17,81 @@ import Discord
 import Discord.Types
 import qualified Discord.Requests as R
 
+-- ####################################################### Core command stuff
+
+{-
+    This is the function that handles any message that starts with '!'
+    but the bot won't respond to commands that are here
+-}
 execCommand :: Message -> DiscordHandler ()
 execCommand msg
-    -- Handle invalid cases
-    | not $ isValidCommand msg = do
-        _ <- restCall $ R.CreateReaction (messageChannel msg, messageId msg) "interrobang"
-        _ <- restCall $ R.CreateMessage (messageChannel msg) "Get it right dickhead" 
-        pure ()
+    -- Ignore Invalid commands
+    | not $ isValidCommand msg = pure ()
     -- Valid commands follow
+    -- Case statement used to associate cmd strings with functions
     | otherwise = case messageCommand msg of
                     "coinflip" -> coinflip msg
 
+
+-- Check if a Message is a command (starts with !)
 isCommand :: Message -> Bool
 isCommand = T.isPrefixOf "!" . T.toLower . messageText
 
+-- Given a message extract the command string !command arg1 arg
 messageCommand :: Message -> T.Text
 messageCommand = T.tail . head . T.words . messageText
 
+-- Given a message extract the arguments
 messageCommandArgs :: Message -> [T.Text]
 messageCommandArgs = tail . T.words . messageText
 
+-- Check if a given message is a valid and registered command
 isValidCommand :: Message -> Bool
 isValidCommand = (`elem` commands) . messageCommand
 
+-- A string list of all commands
 commands :: [T.Text]
 commands = ["coinflip", "8ball"]
 
+-- ####################################################### Command helpers
+
+react :: T.Text -> Message -> DiscordHandler ()
+react emoji msg = (restCall $ R.CreateReaction (messageChannel msg, messageId msg) emoji)
+    >>= \_ -> pure ()
+
+respond :: T.Text -> Message -> DiscordHandler ()
+respond response msg = (restCall $ R.CreateMessage (messageChannel msg) response)
+    >>= \_ -> pure ()
+
+negReact :: Message -> DiscordHandler ()
+negReact = react "interrobang"
+
+posReact :: Message -> DiscordHandler () 
+posReact = react "white_check_mark"
+
```

Comment:
All to avoid having clunky `restCall $ R.Create...` commands in every command block made a bunch of convenient functions to handle the physical response and only take the needed values